pkg/platform/provider/baremetal/cluster/update.go

```diff
@@ -140,6 +146,49 @@ func (p *Provider) EnsurePreClusterUpgradeHook(ctx context.Context, c *v1.Cluste
 	return util.ExcuteCustomizedHook(ctx, c, platformv1.HookPreClusterUpgrade, c.Spec.Machines[:1])
 }
 
+func (p *Provider) EnsureUpgradeCoreDNS(ctx context.Context, c *v1.Cluster) error {
+	if version.Compare(c.Spec.Version, "1.19.0") >= 0 {
```

Comment:
I prefer to depends on  something like 
 https://github.com/tkestack/tke/blob/7ac04217f6c89927e1e86e7d10b3b54ddad5cb58/pkg/util/apiclient/version.go#L29 to handle version